### DISCUSSION TITLE
Transaction of Lesson 3 Challenge(Via QR) is failing again and again.

### DISCUSSION THREAD

**Discussion by heysourin** [Cyfrin/foundry-full-course-cu#22] [Q&A] [answered]
I am seeing this error message. Even after overriding the transaction, its not getting successful. Can someone help? Thank you!! (PS: I am not sure about the 2nd parameter of 'solveChallenge()' function, I have tried with 4 and 2, but facing same error for both the cases.)

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/98945276/8d9dde14-a6c4-4ae9-82b6-064fa899f5c0)

**Comment by cromewar** [comment]
Hello there @heysourin is Metamask shows the error cannot estimate the gas it means you didn't solve the problem yet, but don't give up keep trying!

**Reply by heysourin** [reply]
Hey @cromewar, I didn't understand the meaning of the second parameter, can you please explain?
"* @param newLocation - Pick the spot of the boolean for the next person!"

**Reply by alymurtazamemon** [reply, marked as answer]
@heysourin It means you have to give the index number of the array where the `true` value will be stored and for the next person it will challenge to guess the position of the true value.

**Reply by cromewar** [reply]
@alymurtazamemon nice explanation!

**Reply by G-r-ay** [reply]
Thanks for the explanation!! i had a hard time with it too

**Reply by DuyVo96** [reply]
I stuck with Lesson 7 Challenge. Can you guys give me a hint? Thanks.

**Reply by alymurtazamemon** [reply]
> I stuck with Lesson 7 Challenge. Can you guys give me a hint? Thanks.

@DuyVo96  Do you know how to read private storage variables? It is a hint!!!

**Reply by DuyVo96** [reply]
@alymurtazamemon Ah I got it! Thanks!

**Comment by gulshanpr** [comment]
hey @heysourin , did you get you answer bcoz im getting the same thing

**Reply by heysourin** [reply]
@gulshanpr you mean lesson3 ? nhi hua yrr. aap ka hua toh btana!!

**Reply by gulshanpr** [reply]
Sad, I'll let you know.

**Reply by bagariaraj23** [reply]
@gulshanpr @heysourin As far as i think, 
Location : you have to enter the index value where you think that the bool value is true;
newLocation : you have to give an index as an input.

Hint: read the solveChallenge function and you'll get to know the operations and how the arrays values changes.
Hint2: Check for inherited contracts and "read" them in the etherscan carefully
